pkg/scheduler/framework/plugins/podtopologyspread/filtering.go

```diff
@@ -359,6 +352,13 @@ func (pl *PodTopologySpread) Filter(ctx context.Context, cycleState *framework.C
 			continue
 		}
 
+		selfMatchNum := 0
```

Comment:
Better to add some explanations what this PR for, this is why `Special notes for your reviewer` exists. I guess you want to fail fast here, right?

Reply:
@lokichoggio It's weird to call it "fail fast". I guess you meant "move the definition of variables to their most recent usage"?

Reply:
em...... yes, `fail fast` is a little weird, but I think the most valuable thing here is to prevent unnecessary computing when error in calculating `minMatchNum`. Whatever, `fail fast` is not that accurate.